persist-service/modules/persist-service-ls-extension/src/main/java/io/ballerina/persist/extension/PersistClient.java

```diff
@@ -0,0 +1,641 @@
+        }
+
+        if (selectedTables == null || selectedTables.length == 0) {
+            throw new PersistClientException("Selected tables cannot be null or empty");
+        }
+
+        // If the directory exists with at least one bal file, then throw error for now
+        // Once persist supports multiple model files, this can be removed
+        Path persistPath = this.projectPath.resolve(PERSIST_DIR);
+        if (Files.exists(persistPath)) {
+            try (var files = Files.list(persistPath)) {
+                if (files.anyMatch(path -> path.toString().endsWith(".bal"))) {
+                    throw new PersistClientException("Currently only one database connection is supported per " +
+                            "project. A database connection is already present in the project at: " + persistPath);
+                }
+            } catch (IOException e) {
+                throw new PersistClientException("Error checking existing database connections: " + e.getMessage(), e);
+            }
+        }
+
+        Path persistModelPath = persistPath.resolve(MODEL_FILE_NAME);
+
+        try {
+            Project project = this.workspaceManager.loadProject(projectPath);
+            String packageName = project.currentPackage().packageName().value();
+            Map<Path, List<TextEdit>> textEditsMap = new HashMap<>();
+            boolean isModuleExists = addTextEditForBallerinaToml(packageName, module, textEditsMap);
+            Module entityModule = getInitialEntityModule(selectedTables);
+            SyntaxTree dataModels = new DbModelGenSyntaxTree().getDataModels(entityModule);
+            addTextEditForPersistModelFile(dataModels, textEditsMap, persistModelPath);
+            // Need to reload since the current initial entity module does not have enums
+            // and other details
+            entityModule = reloadEntityFromSyntaxTree(module, dataModels);
+            addTextEditsForClientModuleSources(module, textEditsMap, entityModule);
+            addTextEditForConfigurations(textEditsMap);
+            addTextEditForConnectionClient(packageName, module, textEditsMap);
+            return gson.toJsonTree(new PersistClientResponse(isModuleExists, textEditsMap));
+        } catch (BalException | IOException | FormatterException | WorkspaceDocumentException | EventSyncException e) {
+            throw new PersistClientException("Error introspecting database: " + e.getMessage(), e);
+        }
+    }
+
+    private void addTextEditForConfigurations(Map<Path, List<TextEdit>> textEditsMap) {
+        Path configFilePath = projectPath.resolve("config.bal");
+        boolean isConfigFileExist = Files.exists(configFilePath);
+        Optional<Document> document = isConfigFileExist ? workspaceManager.document(configFilePath)
+                : Optional.empty();
+        String config = LS + "configurable string " + this.name + "Host = \"" + this.host + "\";" + LS;
```

Comment:
if there is already a symbol name like `studentHost` there will be an error.

Reply:
Yes, we need UI support as well to fix those scenarios. Currently, the UI assumes the configurations to be in this format: `<connector-name><config-name>`. During the review we agreed to have a record here instead of separate configurable variables. So I think we can fix this scenario with that improvement. We could put an effort to fix this in UI and LS, but that will completely change when we support a single record typed configurable. Also this is an edge case

Related issue: https://github.com/wso2/product-ballerina-integrator/issues/2123